app/main.py

```diff
@@ -1,3 +1,81 @@
+from __future__ import annotations
+
+
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: int) -> None:
+        self.km = km
+
+    def __add__(self, other: int) -> Distance:
+        if isinstance(other, Distance):
+            return Distance(self.km + other.km)
+        elif isinstance(other, (int, float)):
+            return Distance(self.km + other)
+        else:
+            raise TypeError("Unsupported type for addition of distance")
+
+    def __iadd__(self, other: int) -> Distance:
+        if isinstance(other, Distance):
+            self.km += other.km
+        elif isinstance(other, (int, float)):
+            self.km += other
+        else:
+            raise TypeError("Unsupported type for addition")
+        return self
+
+    def __mul__(self, other: int) -> Distance:
+        if isinstance(other, (int, float)):
+            return Distance(self.km * other)
+        else:
+            raise TypeError("Unsupported type for multiplication")
+
+    def __truediv__(self, divisor: int) -> Distance:
+        if isinstance(divisor, (int, float)):
+            return Distance(round(self.km / divisor, 2))
+        else:
+            raise TypeError("Unsupported type for division")
+
+    def __lt__(self, other: int) -> bool:
+        if isinstance(other, Distance):
+            return self.km < other.km
+        elif isinstance(other, (int, float)):
+            return self.km < other
+        else:
+            raise TypeError("Unsupported type for comparison")
+
+    def __gt__(self, other: int) -> bool:
+        if isinstance(other, Distance):
+            return self.km > other.km
+        elif isinstance(other, (int, float)):
+            return self.km > other
+        else:
+            raise TypeError("Unsupported type for comparison")
+
+    def __eq__(self, other: int) -> bool:
```

Comment:
The type annotation for the `other` parameter should be `Union[Distance, int, float]` instead of just `int` to accurately reflect the types that can be compared with a `Distance` object.